### ISSUE TITLE
java.util.ConcurrentModificationException  com.telerik.widget.list.ListViewGestureListener.onTapUp(ListViewGestureListener.java:165)

### ISSUE THREAD

**Issue by avivanco** [telerik/nativescript-ui-feedback#538]
### Did you verify this is a real problem by searching [Stack Overflow](http://stackoverflow.com/questions/tagged/nativescript)?
No

### Tell us about the problem
Problem when I select an item  of my RadListView from an angular ModalDialogService.

### Which platform(s) does your issue occur on?
Android

### Please provide the following version numbers that your issue occurs with:
- **Progress NativeScript UI version:** 3.4.0
- **CLI:** 3.4.0
- **Cross-platform modules:** 3.4.0
- **Runtime(s):** `"tns-android"`  3.4.1

### Please tell us how to recreate the issue in as much detail as possible.
I'm showing a RadListView from an angular ModalDialogService, and when I select an item of my RadListView I'm passing the information to main page, but I'm getting this error:

This error occurs with the last version of Progress NativeScript UI 3.4.0, with 3.3.0 works fine.

System.err: java.util.ConcurrentModificationException
System.err:     at java.util.ArrayList$Itr.next(ArrayList.java:860)
System.err:     at com.telerik.widget.list.ListViewGestureListener.onTapUp(ListViewGestureListener.java:165)
System.err:     at com.telerik.widget.list.ListViewGestureListener.onSingleTapUp(ListViewGestureListener.java:242)
System.err:     at android.view.GestureDetector.onTouchEvent(GestureDetector.java:640)
System.err:     at com.telerik.widget.list.ListViewGestureListener.onTouchEvent(ListViewGestureListener.java:145)
System.err:     at com.telerik.widget.list.RadListView.onTouchEvent(RadListView.java:542)
System.err:     at android.view.View.dispatchTouchEvent(View.java:11725)
System.err:     at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2955)
System.err:     at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2636)
System.err:     at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2961)
System.err:     at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2650)
System.err:     at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2961)
System.err:     at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2650)
System.err:     at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2961)
System.err:     at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2650)
System.err:     at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2961)
System.err:     at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2650)
System.err:     at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2961)
System.err:     at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2650)
System.err:     at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2961)
System.err:     at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2650)
System.err:     at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2961)
System.err:     at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2650)
System.err:     at com.android.internal.policy.DecorView.superDispatchTouchEvent(DecorView.java:445)
System.err:     at com.android.internal.policy.PhoneWindow.superDispatchTouchEvent(PhoneWindow.java:1828)
System.err:     at android.app.Dialog.dispatchTouchEvent(Dialog.java:815)
System.err:     at com.android.internal.policy.DecorView.dispatchTouchEvent(DecorView.java:407)
System.err:     at android.view.View.dispatchPointerEvent(View.java:11964)
System.err:     at android.view.ViewRootImpl$ViewPostImeInputStage.processPointerEvent(ViewRootImpl.java:4776)
System.err:     at android.view.ViewRootImpl$ViewPostImeInputStage.onProcess(ViewRootImpl.java:4590)
System.err:     at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:4128)
System.err:     at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:4181)
System.err:     at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:4147)
System.err:     at android.view.ViewRootImpl$AsyncInputStage.forward(ViewRootImpl.java:4274)
System.err:     at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java:4155)
System.err:     at android.view.ViewRootImpl$AsyncInputStage.apply(ViewRootImpl.java:4331)
System.err:     at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:4128)
System.err:     at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:4181)
System.err:     at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:4147)
System.err:     at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java:4155)
System.err:     at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:4128)
System.err:     at android.view.ViewRootImpl.deliverInputEvent(ViewRootImpl.java:6642)
System.err:     at android.view.ViewRootImpl.doProcessInputEvents(ViewRootImpl.java:6616)
System.err:     at android.view.ViewRootImpl.enqueueInputEvent(ViewRootImpl.java:6577)
System.err:     at android.view.ViewRootImpl$WindowInputEventReceiver.onInputEvent(ViewRootImpl.java:6745)
System.err:     at android.view.InputEventReceiver.dispatchInputEvent(InputEventReceiver.java:185)
System.err:     at android.os.MessageQueue.nativePollOnce(Native Method)
System.err:     at android.os.MessageQueue.next(MessageQueue.java:325)
System.err:     at android.os.Looper.loop(Looper.java:142)
System.err:     at android.app.ActivityThread.main(ActivityThread.java:6541)
System.err:     at java.lang.reflect.Method.invoke(Native Method)
System.err:     at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:240)
System.err:     at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:767)

### Is there code involved? If so, please share the minimal amount of code needed to recreate the 
Select method I'm using is: 
With version 3.3.0 this works fine. The problem is with 3.4.0
public onItemSelected(args) {
        const listview = args.object as RadListView;
        const selectedItems = listview.getSelectedItems() as Array<any>;
        this.modal.closeCallback(selectedItems[0]);
    }

**Comment by tigrenok00** [comment]
This happens for us as well with the newest version of the separate listview plugin:

```
{
  "description": "NativeScript Application",
  "license": "SEE LICENSE IN <your-license-filename>",
  "readme": "NativeScript Application",
  "repository": "<fill-your-repository-here>",
  "nativescript": {
    "id": "XXX",
    "tns-android": {
      "version": "3.2.0"
    },
    "tns-ios": {
      "version": "3.2.0"
    }
  },
  "dependencies": {
    "@angular/common": "^4.3.6",
    "@angular/compiler": "^4.3.6",
    "@angular/core": "^4.3.6",
    "@angular/forms": "^4.3.6",
    "@angular/http": "^4.3.6",
    "@angular/platform-browser": "^4.3.6",
    "@angular/platform-browser-dynamic": "^4.3.6",
    "@angular/router": "^4.2.6",
    "nativescript-angular": "^4.2.0",
    "nativescript-angular-snapshot": "3.0.0-5.5.372.32",
    "nativescript-animatecss": "^1.0.0",
    "nativescript-dom": "^1.0.8",
    "nativescript-floatingactionbutton": "^3.0.1",
    "nativescript-globalevents": "^1.1.0",
    "nativescript-loading-indicator": "^2.3.2",
    "nativescript-ngx-fonticon": "^2.2.0",
    "nativescript-pro-ui": "^3.2.0",
    "nativescript-screen-orientation": "^2.0.0",
    "nativescript-theme-core": "1.0.4",
    "reflect-metadata": "~0.1.8",
    "rxjs": "~5.3.0",
    "tns-core-modules": "^3.2.0"
  },
  "devDependencies": {
    "babel-traverse": "6.4.5",
    "babel-types": "6.4.5",
    "babylon": "6.4.5",
    "lazy": "1.0.11",
    "nativescript-dev-android-snapshot": "^0.*.*",
    "nativescript-dev-typescript": "^0.4.4",
    "typescript": "~2.3.2",
    "zone.js": "~0.8.2"
  }
}

```

**Comment by kelvinliuhk** [comment]
I got same error, 
but I using "nativescript-pro-ui": "^3.4.0",
others packages version same with you.
My Android device is version 7.1.2 
No idea how to fix it.

**Comment by avivanco** [comment]
They changed nativescript-pro-ui dependency, that is not anymore, you should use this:

"nativescript-ui-autocomplete": "^3.5.0",
    "nativescript-ui-calendar": "^3.5.0",
    "nativescript-ui-chart": "^3.5.0",
    "nativescript-ui-dataform": "^3.5.0",
    "nativescript-ui-gauge": "^3.5.0",
    "nativescript-ui-listview": "^3.5.0",
    "nativescript-ui-sidedrawer": "^3.5.0",

with that I didn't get the error again.

Hope it helps.

**Comment by NickIliev** [comment]
Related to [https://github.com/telerik/nativescript-ui-feedback/issues/530](https://github.com/telerik/nativescript-ui-feedback/issues/530)

**Comment by zbranzov** [comment]
Should be fixed in the latest nativescript-ui-listview 3.5.7
